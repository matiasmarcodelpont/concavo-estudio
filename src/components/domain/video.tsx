import { ChevronRight } from 'lucide-react'
import Link from 'next/link'
import Heading from './Heading'

interface VideoProps {
  src: string
  title: string
  href: string
}

export const Video = ({ src, title, href }: VideoProps) => {
  return (
    <Link href={href} className='block relative w-full h-[80vh] overflow-hidden'>
      <video
        src={src}
        title={title}
        className='absolute top-0 left-0 w-full h-full border-0 pointer-events-none z-10 border-none object-cover'
        height='100%'
        autoPlay
        muted
        loop
      />
      <div className='absolute inset-0 bg-black opacity-50 z-20' />
      <div className='absolute inset-0 flex flex-col z-30 justify-center'>
        <Heading className='text-white uppercase mb-0 text-3xl md:text-4xl'>{title}</Heading>
        <p className='text-white text-lg md:text-2xl'>Lorem ipsum dolor sit.</p>
        {href && (
          <ChevronRight
            className='absolute top-1/2 right-2 p-4 rounded-full translate-y-[-50%] hidden md:block'
            size='84px'
            color='white'
          />
        )}
      </div>
    </Link>
  )
}
