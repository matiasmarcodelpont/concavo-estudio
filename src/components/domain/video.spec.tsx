import '@testing-library/jest-dom'
import { render } from '@testing-library/react'

import { Video } from './video'

describe('Video component', () => {
  const videoProps = {
    src: 'https://www.example.com/video',
    title: 'cuchuflito',
    href: 'fake-link',
  }

  it('matches snapshot', () => {
    const { container } = render(<Video {...videoProps} />)
    expect(container).toMatchSnapshot()
  })

  it('renders an iframe with the correct src', () => {
    const component = render(<Video {...videoProps} />)

    const iframe = component.getByTitle('cuchuflito')
    expect(iframe).toHaveAttribute('src', videoProps.src)
    expect(iframe).toHaveAttribute('allow', 'autoplay; fullscreen; picture-in-picture; clipboard-write')
  })

  it('renders a link if a href prop is passed', () => {
    const component = render(<Video {...videoProps} />)

    const link = component.getByRole('link')
    expect(link).toHaveAttribute('href', videoProps.href)
  })
})
