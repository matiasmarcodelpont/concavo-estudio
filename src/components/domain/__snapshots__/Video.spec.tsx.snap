// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Video component matches snapshot 1`] = `
<div>
  <a
    class="block relative w-full h-[80vh] overflow-hidden"
    href="fake-link"
  >
    <video
      autoplay=""
      class="absolute top-0 left-0 w-full h-full border-0 pointer-events-none z-10 border-none object-cover"
      height="100%"
      loop=""
      src="https://www.example.com/video"
      title="cuchuflito"
    />
    <div
      class="absolute inset-0 bg-black opacity-50 z-20"
    />
    <div
      class="absolute inset-0 flex flex-col z-30 justify-center items-center"
    >
      Lorem, ipsum dolor sit amet consectetur adipisicing elit
      <div
        class="absolute bottom-4 right-1/2 p-2 md:p-4 translate-x-[50%] flex flex-col items-center gap-2 text-bone text-sm sm:text-base md:text-lg"
      >
        START HERE
      </div>
    </div>
  </a>
</div>
`;
